internal/datacoord/session_manager.go

```diff
@@ -521,11 +522,23 @@ func (c *SessionManagerImpl) CheckHealth(ctx context.Context) error {
 			}
 
 			sta, err := cli.GetComponentStates(ctx, &milvuspb.GetComponentStatesRequest{})
-			if err != nil {
+			if err = merr.CheckRPCCall(sta, err); err != nil {
+				return fmt.Errorf("GetComponentStates fails for datanode:%d, %w", nodeID, err)
```

Comment:
doesn't make sense to have two calls here.

We can analyze states on checkHealth call